src/plugin/src/main/java/com/defenseunicorns/uds/keycloak/plugin/clientpolicy/executor/UDSClientPolicyPermissionsExecutorFactory.java

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright 2024 Defense Unicorns
+ * SPDX-License-Identifier: AGPL-3.0-or-later OR LicenseRef-Defense-Unicorns-Commercial
+ */
+
+package com.defenseunicorns.uds.keycloak.plugin.clientpolicy.executor;
+
+import org.keycloak.Config;
+import org.keycloak.models.KeycloakSession;
+import org.keycloak.models.KeycloakSessionFactory;
+import org.keycloak.provider.ProviderConfigProperty;
+import org.keycloak.services.clientpolicy.executor.ClientPolicyExecutorProviderFactory;
+
+import java.util.Collections;
+import java.util.List;
+
+/**
+ * Keycloak Client Policy Executor Factory for the UDS Operator.
```

Comment:
it wasnt immediately clear to me what the use of this was for and had to go dig through the keycloak docs to understand it a bit more. i think it would be good to add some more context here. 
```suggestion
 * Factory for creating and registering the UDS Client Policy Executor with Keycloak.
 *
 * This class is needed to integrate the custom UDS security hardening logic into Keycloak's
 * client management process. It creates instances of UDSClientPolicyPermissionsExecutor, which
 * enforce policies such as:
 * - Marking clients as created by the UDS Operator by setting the "created-by=uds-operator" attribute.
 * - Restricting operations on clients not owned by the UDS Operator or not following the naming convention.
 * - Disabling the "Full Scope Allowed" feature and limiting client scopes to an allowed set.
 *
 * Without this factory, Keycloak would not be able to instantiate and manage the custom policy
 * executor, and the UDS-specific client policies would not be enforced.
```